src/VolumeLoader.ts

```diff
@@ -104,4 +140,319 @@ export default class VolumeLoader {
+    // update levelToLoad after we get size info about multiscales?
+
+    const metadata = allmetadata.omero;
+
+    const level0 = await openArray({ store: store, path: imagegroup + "/" + dataset0.path, mode: "r" });
+    // full res info
+    const w = level0.meta.shape[4];
+    const h = level0.meta.shape[3];
+    const z = level0.meta.shape[2];
+    const c = level0.meta.shape[1];
+    const sizeT = level0.meta.shape[0];
+    console.log(`X=${w}, Y=${h}, Z=${z}, C=${c}, T=${sizeT}`);
+
+    // making a choice of a reduced level:
+    const downsampleZ = 2; // half the z
+    const levelToLoad = numlevels - 1; //1;
+    const dataset2 = allmetadata.multiscales[imageIndex].datasets[levelToLoad];
+    const level = await openArray({ store: store, path: imagegroup + "/" + dataset2.path, mode: "r" });
+
+    // reduced level info
+    const tw = level.meta.shape[4];
+    const th = level.meta.shape[3];
+
+    // compute rows and cols and atlas width and ht, given tw and th
+    const loadedZ = Math.ceil(z / downsampleZ);
+    const { nrows, ncols } = computePackedAtlasDims(loadedZ, tw, th);
+    const atlaswidth = ncols * tw;
+    const atlasheight = nrows * th;
+    console.log(atlaswidth, atlasheight);
+
+    const chnames: string[] = [];
+    for (let i = 0; i < metadata.channels.length; ++i) {
+      chnames.push(metadata.channels[i].label);
+    }
+    /* eslint-disable @typescript-eslint/naming-convention */
+    const imgdata: ImageInfo = {
+      width: w,
+      height: h,
+      channels: c,
+      channel_names: chnames,
+      rows: nrows,
+      cols: ncols,
+      tiles: loadedZ,
+      tile_width: tw,
+      tile_height: th,
+      // for webgl reasons, it is best for atlas_width and atlas_height to be <= 2048
+      // and ideally a power of 2.  This generally implies downsampling the original volume data for display in this viewer.
+      atlas_width: atlaswidth,
+      atlas_height: atlasheight,
+      pixel_size_x: scale5d[4],
+      pixel_size_y: scale5d[3],
+      pixel_size_z: scale5d[2] * downsampleZ,
+      name: metadata.name,
+      version: metadata.version,
+      transform: {
+        translation: [0, 0, 0],
+        rotation: [0, 0, 0],
+      },
+      times: sizeT,
+    };
+    /* eslint-enable @typescript-eslint/naming-convention */
+
+    // got some data, now let's construct the volume.
+    const vol = new Volume(imgdata);
+
+    const storepath = imagegroup + "/" + dataset2.path;
+    // do each channel on a worker
+    for (let i = 0; i < c; ++i) {
+      const worker = new Worker(new URL("./workers/FetchZarrWorker", import.meta.url));
+      worker.onmessage = function (e) {
+        const u8 = e.data.data;
+        const channel = e.data.channel;
+        console.log("begin setchannel and callback");
+        vol.setChannelDataFromVolume(channel, u8);
+        if (callback) {
+          // make up a unique name? or have caller pass this in?
+          callback(urlStore + "/" + imageName, vol, channel);
```

Comment:
Might call this something like `onChannelLoaded` - having a field called `callback` on a promise-based function confused me for a moment. It seems reasonable to me to say that loading a single channel is an event that happens while `loadZarr`/`loadTiff` is running rather than a completed action that gets a callback, even if internally it means a worker has finished.

Reply:
good idea, I'll rename that